book/src/dev/ecc-updates.md

```diff
@@ -0,0 +1,68 @@
+# Updating the ECC dependencies
+
+Zebra relies on numerous Electric Coin Company ([ECC](https://electriccoin.co/)) dependencies, and updating them can be a complex task. This guide will help you navigate the process.
+
+
+The main dependency that influences that is [zcash](https://github.com/zcash/zcash) itself. This is because [zebra_script](https://github.com/ZcashFoundation/zcash_script) links to specific files from it (zcash_script.cpp and all on which it depends). Due to the architecture of zcash, this requires linking to a lot of seemingly unrelated dependencies like orchard, halo2, etc (which are all Rust crates).
+
+## Steps for upgrading
+
+Let's dive into the details of each step required to perform an upgrade:
+
+### Before starting
+
+- Zebra developers often dismiss ECC dependency upgrade suggestions from dependabot. For instance, see [this closed PR](https://github.com/ZcashFoundation/zebra/pull/7745) in favor of the [5.7.0 zcashd upgrade PR](https://github.com/ZcashFoundation/zebra/pull/7784), which followed this guide.
+
+- Determine the version of `zcashd` to use. This version will determine which versions of other crates to use. Typically, this should be a [tag](https://github.com/zcash/zcash/tags), but in some cases, it might be a reference to a branch (e.g., nu5-consensus) for testing unreleased developments.
+
+- Upgrading the `zcash_script` crate can be challenging, depending on changes in the latest `zcashd` release. Follow the instructions in the project's [README](https://github.com/ZcashFoundation/zcash_script/blob/master/README.md) for guidance.
+
+- If you haven't upgraded `zcash_script`, don't proceed with upgrading other ECC dependencies in Zebra.
+
+- Ensure that the crate versions in the `Cargo.toml` of the zcashd release, `Cargo.toml` of `zcash_script`, and the `Cargo.toml` files of Zebra crates are all the same. Version consistency is crucial.
```

Comment:
When is this version consistency check done?
If it is done before starting, then it will always be true, because nothing has been upgraded yet.

Does it belong at the top of the "Upgrade versions" section instead?

Reply:
https://github.com/ZcashFoundation/zebra/pull/7794/commits/84ba7c728c47d97523af9c425e1c3ca850079792